platform/database/tests/helpers.go

```diff
@@ -0,0 +1,69 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/GuiaBolso/darwin"
+	"github.com/fewlinesco/go-pkg/platform/database"
+	"github.com/jmoiron/sqlx"
+)
+
+func migrate(cfg database.Config, t *testing.T) func() {
+	db, err := database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to DB: %#v, with config: %#v", err, cfg)
+	}
+	defer db.Close()
+
+	err = database.Migrate(db, []darwin.Migration{
+		{
+			Version:     1,
+			Description: "Create test data table",
+			Script: `
+				DROP TABLE IF EXISTS test_data;
+				CREATE TABLE test_data(
+					id UUID PRIMARY KEY,
+					code VARCHAR(63)
+				)`,
+		},
+	})
+
+	if err != nil {
+		t.Fatalf("could not migrate the database: %#v", err)
+	}
+
+	return func() {
+		db, err := database.Connect(cfg)
+		if err != nil {
+			t.Fatalf("could not connect to DB: %#v", err)
+		}
+		defer db.Close()
+
+		_, err = db.ExecContext(context.Background(), `DROP TABLE IF EXISTS test_data; DROP TABLE IF EXISTS darwin_migrations;`)
+		if err != nil {
+			t.Fatalf("could not clean the database: %#v", err)
+		}
+	}
+}
+
+func connect(config database.Config) (*sqlx.DB, error) {
+	if config.URL == "" {
+		options := make(url.Values)
+		for key, value := range config.Options {
+			options.Add(key, value)
+		}
+
+		connectionURL := url.URL{
+			Scheme:   config.Scheme,
+			User:     url.UserPassword(config.Username, config.Password),
+			Host:     fmt.Sprintf("%s:%d", config.Host, config.Port),
+			Path:     config.Database,
+			RawQuery: options.Encode(),
+		}
+		config.URL = connectionURL.String()
+	}
+	return sqlx.Connect(config.Driver, config.URL)
+}
```

Comment:
Don't we already have this function for the main DB?

Reply:
it's private and I didn't want to give a public access to a function returning a sqlx type

Reply:
Can't we avoid calling `connect()` and just use the exported API for the tests here ?

Reply:
what kind of value would this add to paste this code everywhere ?

Reply:
like I understand the need to use public code for what we test. But this is purposedly not exposed publicly, and pasting it in test would add noise imho.

Reply:
I'm not saying we should paste this everywhere, I was wondering if we could make the tests work without having to use a `sqlx.DB` directly at all in the tests.

Reply:
Sorry my first comment wasn't clear and you answered while I was editing it to add details 🙇

Reply:
Ok but this was actually discussed last week and we decided use sqlx because otherwise how would you be sure it's not a false positive / negative from the package itself ?